### ISSUE TITLE
[BUG] indexing.quadtree_on_points causes memory access error

### ISSUE THREAD

**Issue by khoaxtru20** [rapidsai/cuspatial#302]
**Describe the bug**

Sometimes when I execute the `indexing.quadtree_on_points` function, I get this error:
`RuntimeError: reduce failed to synchronize: cudaErrorIllegalAddress: an illegal memory access was encountered`

**Steps/Code to reproduce bug**
Follow this guide http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports to craft a minimal bug report. This helps us reproduce the issue you're having and resolve the issue more quickly.

I basically use the example from the documentation - same variable assignments for scale, max_depth, and min_size - except I'm using negative/positive longitude and latitude points. I'm looping over a large dataset in chunks, and it doesn't seem to matter what chunksize I choose, the error will occur randomly. Sometimes it will happen after 2 calls, sometimes it will happen after 20.

Here is the Jupyter notebook cell that creates the error:
```
# Getting chunks one at a time, trying to pinpoint problem (same error occurs in a loop)
vf_cudf = vf_df_reader.get_chunk()

# Only interested in the chunk's area
box = create_bounding_box_of_interest(vf_cudf["longitude"].min(),
                                      vf_cudf["longitude"].max(),
                                      vf_cudf["latitude"].min(),
                                      vf_cudf["latitude"].max())
print('Created box')

# Create spatial index
key_to_point, quadtree = cusp.quadtree_on_points(
    vf_cudf["longitude"],
    vf_cudf["latitude"],
    box['x_min'], box['x_max'],
    box['y_min'], box['y_max'],
    scale, max_depth, min_size
)

# Error is produced before the following print lines, execution is stopped here
print('Created ktp:', len(key_to_point))
print('Created quadtree:', len(quadtree))

# Link points to bounding boxes
poly_quad_pairs = cusp.join_quadtree_and_bounding_boxes(quadtree, bboxes,
                                                    box['x_min'], box['x_max'],
                                                    box['y_min'], box['y_max'],
                                                    scale, max_depth)
print('Created poly quad pairs', len(poly_quad_pairs))
     
# Find points in polygons
sjoineddf = cusp.quadtree_point_in_polygon(poly_quad_pairs, quadtree, key_to_point,
                                       vf_cudf["longitude"], vf_cudf["latitude"],
                                       poly_offsets, ring_offsets,
                                       points_cudf.x, points_cudf.y)
print('Joined point and polygon')

# Some code to join DataFrames
sjoineddf['vf_index'] = cudf.Series(key_to_point.iloc[sjoineddf['polygon_index']],index=np.arange(0,len(sjoineddf)))

pd_sjoineddf = sjoineddf.to_pandas()
pd_sjoineddf = pd_sjoineddf.join(sf_precincts,on='point_index')
pd_sjoineddf = pd_sjoineddf.drop(columns=['geometry','point_index', 'polygon_index'])

vf_cudf['vf_index'] = np.arange(0,len(vf_cudf))
vf_cudf = vf_cudf.join(pd_sjoineddf.set_index('vf_index'), how="left", on='vf_index', lsuffix='_vf')
vf_cudf = vf_cudf.drop(columns='vf_index')

linked_df = linked_df.append(vf_cudf)
```
**Expected behavior**

The `indexing.quadtree_on_points` should reliably execute without memory access problems

**Environment details (please complete the following information):**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuSpatial install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used

I'm running a Jupyter Notebook on Gradient Papersource provisioned with the most recent rapidsai/rapidsai Docker container. The machine uses a Quadro P5000 GPU.

**Additional context**

This is my first issue posting so let me know how I can clarify my situation.

**Comment by drabastomek** [comment]
This might have been addressed in #320 cc: @kkraus14 @harrism

**Comment by kkraus14** [comment]
Fixed by #320
